### PR TITLE
add init defaultCMK for KMS service

### PR DESCRIPTION
**Pull request by Yidingluo1**
When the KMS service is started, the defaultCMK will be created, if it
has been created before, success will be returned.
test: passed the test

Signed-off-by: luoyiding <luo.yd@neusoft.com>

### CODE REVIEW

File: ehsm_kms_service/secret_management.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    const napi_res = JSON.parse(ehsm_napi[`NAPI_${cryptographic_apis.CreateKey}`](keyspec, origin))
+                    const creationDate = new Date().getTime()
+                    DB.insert({
+                        _id: `cmk:${keyid}`,
+                        keyid,
+                        keyBlob: napi_res.result.cmk,
+                        creator: 'System',
+                        creationDate,
+                        expireTime: creationDate + CMK_EFFECTIVE_DURATION * 999,
+                        alias: 'System Default CMK',
```
Comment (syan10) [inline ehsm_kms_service/secret_management.js:41]: 'Secret Manager Default CMK'

File: ehsm_kms_service/secret_management.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                if (cmks_res.docs.length == 0) {
+                    const keyspec = 0
+                    const origin = 0
+                    const napi_res = JSON.parse(ehsm_napi[`NAPI_${cryptographic_apis.CreateKey}`](keyspec, origin))
+                    const creationDate = new Date().getTime()
+                    DB.insert({
+                        _id: `cmk:${keyid}`,
+                        keyid,
+                        keyBlob: napi_res.result.cmk,
+                        creator: 'System',
```
Comment (syan10) [inline ehsm_kms_service/secret_management.js:38]: you can change the "System" to "Secret Manager"

File: ehsm_kms_service/constant.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ const CMK_EFFECTIVE_DURATION = 1 * 365 * 24 * 60 * 60 * 1000
 const CMK_LOOP_CLEAR_TIME = 24 * 60 * 60 * 1000
 const CMK_EXPIRE_TIME_EXPAND = 10 * 24 * 60 * 60 * 1000
 const CMK_LOOP_CLEAR_EXECUTION_TIME = 3
+const KEYID_SYSTEM_DEFAULT_CMK = 'systemDefCMK'
```
Comment (syan10) [inline ehsm_kms_service/constant.js:9]: same here, SYSTEM to SecretManager

Comment (yang8621) [inline ehsm_kms_service/constant.js:9]: Def means define, change to sysDftCMK or others

File: ehsm_kms_service/ehsm_kms_server.js [hunk 4]
```diff
@@ -15,6 +15,9 @@ const {
   _nonce_cache_timer,
   _cmk_cache_timer,
 } = require('./function')
+const {
+  init_DefaultCMK,
```
Comment (syan10) [inline ehsm_kms_service/ehsm_kms_server.js:19]: Init_SM_DefaultCMK

File: ehsm_kms_service/secret_management.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    try {
+        const keyid = KEYID_SYSTEM_DEFAULT_CMK
+        const query = {
+            selector: {
+                keyid,
+            },
+            fields: ['keyid',],
+            limit: 1,
+        }
+        DB.partitionedFind('cmk', query)
```
Comment (yang8621) [inline ehsm_kms_service/secret_management.js:27]: can reuse store_cmk() ?

Comment (Yidingluo1) [reply ehsm_kms_service/secret_management.js:27]: Parameters and return values are different, if reused，store_cmk() needs to be modified

Comment (yang8621) [reply ehsm_kms_service/secret_management.js:27]: OK.

**Comment by yang8621** [comment]
We need separate cmk for different users, but not a default key for all users which introduce security concern.

**Comment by Yidingluo1** [comment]
> We need separate cmk for different users, but not a default key for all users which introduce security concern.

When creating secret, the default CMK will be used only if the user does not fill in the CMK. This default CMK is only called when the service is started, and it is only a default value when there is no CMK to type

**Comment by yang8621** [comment]
> > We need separate cmk for different users, but not a default key for all users which introduce security concern.
> 
> When creating secret, the default CMK will be used only if the user does not fill in the CMK. This default CMK is only called when the service is started, and it is only a default value when there is no CMK to type

So is it possible that two users share the same default CMK? The default CMK still requires being different for different users. That means default CMK should bind to appid.

**Comment by Yidingluo1** [comment]
Scheme abandonment

**Pull request closed without merging**